microsoft/testsuites/cpu/cpusuite.py

```diff
@@ -51,3 +61,69 @@ class CPUSuite(TestSuite):
     )
     def verify_cpu_hot_plug(self, log: Logger, node: Node) -> None:
         verify_cpu_hot_plug(log, node)
+
+    @TestCaseMetadata(
+        description="""
+            This test will check cpu hotplug with storage workload.
+            The cpu hotplug steps are same as `verify_cpu_hot_plug` test case.
+            """,
+        priority=3,
+        requirement=simple_requirement(
+            min_core_count=32,
+        ),
+    )
+    def verify_cpu_offline_storage_workload(self, log: Logger, node: Node) -> None:
+        # run fio process asynchronously on the node and give it 10 seconds to start
+        fio_storage_workload_script = CustomScriptBuilder(
+            Path(__file__).parent.joinpath("scripts"), ["fio_storage_worload.sh"]
+        )
+        process = node.tools[Fio].run_script_async(fio_storage_workload_script)
+        time.sleep(10)
+
+        # verify cpu hotplug functionality
+        verify_cpu_hot_plug(log, node)
+
+        # verify that the fio was running when hotplug was triggered
+        assert_that(
+            process.is_running(), "Storage workload was not running during CPUhotplug"
+        ).is_true()
+
+    @TestCaseMetadata(
+        description="""
+            This test will check cpu hotplug with network workload.
+            The cpu hotplug steps are same as `verify_cpu_hot_plug` test case.
+            """,
+        priority=3,
+        requirement=simple_requirement(
+            min_core_count=32,
+            min_count=2,
+        ),
+    )
+    def verify_cpu_offline_network_workload(
+        self, log: Logger, environment: Environment
+    ) -> None:
+        server = cast(RemoteNode, environment.nodes[0])
+        client = cast(RemoteNode, environment.nodes[1])
+
+        # run iperf3 process asynchronously on the server and client,
+        # and give it 10 seconds to start
+        # server.tools[Iperf3].run_as_server_async()
+        fio_network_client_workload_script = CustomScriptBuilder(
+            Path(__file__).parent.joinpath("scripts"),
+            ["fio_network_client_workload.sh"],
+        )
+        client_iperf3_process = client.tools[Iperf3].run_script_async(
+            fio_network_client_workload_script,
+            parameters=f"{server.internal_address}",
+            shell=True,
+        )
+        time.sleep(10)
```

Comment:
don't hardcode sleep

Reply:
The program needs some setup time before we can run tests. This is something that was present in the v2 tests

Reply:
we can enrich the logic in `iperf3.run_as_client_async` to check if the the networking throughput is up, and then return. If it just needs to wait the process complete starts, it can put a 0.5 sleep in the `run_as_client_async` and comment it.

Reply:
So for fio we need to add atleast 30 seconds of sleep. This is because the tests are not actually started until it creates file which is approx 1GB in size. Should we add this sleep in fio or terstcase?

Reply:
Sleep 30 seconds is terrible. It's not enough sometimes.
1. Reduce the test data size to 100M. It will speed up the test case.
2. check if there is stdout from the process. See if it's possible to check the fio starts to run.